compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -739,13 +739,16 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
         let tcx = self.infcx.tcx;
         // Try to find predicates on *generic params* that would allow copying `ty`
         let infcx = tcx.infer_ctxt().build();
-        if infcx
-            .type_implements_trait(
-                tcx.lang_items().clone_trait().unwrap(),
-                [tcx.erase_regions(ty)],
-                self.param_env,
-            )
-            .must_apply_modulo_regions()
+        let clone_trait_def = tcx.lang_items().clone_trait();
+
+        if clone_trait_def.is_some()
+            && infcx
+                .type_implements_trait(
+                    clone_trait_def.unwrap(),
```

Comment:
I'd prefer if you don't use `Option::is_some` and `Option::unwrap`. Can you either use a `let else` or `if let` to match on `Some`?

Reply:
Sure, will do but why is that?

Reply:
It's not particularly idiomatic to test `is_some` and then unwrap afterwards when we have destructuring syntax built into the language :sweat_smile:

Reply:
Makes sense, done.